### PR TITLE
[TT-7698] Validate OAS after migrate and fill

### PR DESCRIPTION
**Pull request by furkansenharputlu**

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4680/merge
Commit: 8b9ca298b91b2f1a60207cf0fa6b6275905bb6ab 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993099119)

File: apidef/oas/oas.go [hunk 1]
```diff
@@ -373,22 +375,40 @@ func MigrateAndFillOAS(api *apidef.APIDefinition) (APIDef, []APIDef, error) {
 		return baseAPIDef, nil, err
 	}
 
+	baseAPIDef.OAS, err = newOASFromClassicAPIDefinition(api)
+	if err != nil {
+		return baseAPIDef, nil, fmt.Errorf("base API %s migrated OAS is not valid: %s", api.Name, err)
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/oas/oas.go:380]: Actually, I try to say the result OAS is not valid. The migration happened, it is done but the end result is not valid. Maybe another way we can say?

[the remark this answers is not in the document]
Comment (jeffy-mathew) [reply apidef/oas/oas.go:380]: I think if that's the intention we can show where the validation breaks.

**Review by jeffy-mathew — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4680/merge
Commit: dbe2c2915b6ce872b7fc3bb8262ada48acbdc6d1 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993364394)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/4680/merge
Commit: c04da38f61b78f7dd69a91b409b4840dac696461 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/3993359989)

**Pull request merged**